### PR TITLE
BUGFIX: Fix navigation in tree after an in-page navigation

### PR DESCRIPTION
**Pull request by skurfuerst**
## How to reproduce

- Go to page A in document tree
- navigate to page B in content area
- select page A in document tree again
- expected: selection to page A works
- actual: no navigation is triggered to PageA

## Bugfix

- ensure the redux state is always up to date
- do not trigger page refreshes if the URL is already correct in the iFrame

Additionally, the loading indicator does not need to be handled through SET_SRC
anymore, as this is now implemented differently (so that's a code cleanup).

### CODE REVIEW

File: packages/react-ui-components/src/Frame/frame.js [hunk 1]
```diff
@@ -13,6 +14,18 @@ export default class Frame extends PureComponent {
...
         this.ref = ref;
+    };
+
+    componentDidMount() {
+        this.componentDidUpdate();
+    }
+
+    // we do not use react's magic to change to a different URL in the iFrame, but do it explicitely (in order to avoid reloads if we are already on the correct page)
+    componentDidUpdate() {
+        const win = ReactDOM.findDOMNode(this).contentWindow; // eslint-disable-line react/no-find-dom-node
```
Comment (dimaip) [inline packages/react-ui-components/src/Frame/frame.js]: why not use a ref instead of findDOMNode?

Comment (skurfuerst) [reply packages/react-ui-components/src/Frame/frame.js]: would be also fine. I can change it to ref.

File: packages/react-ui-components/src/Frame/frame.js [hunk 2]
```diff
@@ -13,6 +14,22 @@ export default class Frame extends PureComponent {
 
     handleReference = ref => {
         this.ref = ref;
+    };
+
+    componentDidMount() {
+        this.componentDidUpdate();
```
Comment (dimaip) [inline packages/react-ui-components/src/Frame/frame.js]: Please not like this! Extract the logic that should be executed both on mount and update into another method and call it from two places.

Comment (skurfuerst) [reply packages/react-ui-components/src/Frame/frame.js]: ok if you like ^^

**Pull request merged**